pcm.go

```diff
@@ -199,7 +200,17 @@ func (w *sampleWriter[T]) Close() error {
 func (w *sampleWriter[T]) WriteSample(in T) error {
 	data := make([]byte, len(in))
 	copy(data, in)
-	return w.w.WriteSample(media.Sample{Data: data, Duration: w.sampleDur})
+
+	var droppedPackets uint16
+	if !w.lastWrite.IsZero() {
+		timeSinceLastWrite := time.Since(w.lastWrite)
+		if timeSinceLastWrite > w.sampleDur {
```

Comment:
Maybe need to make this a bit more fuzzy? As it is written here, a write coming through after 19.99ms will not insert a silence packet in between. Guess, it is okay, but just something to thinking about a bit and see if some fuzz factor should be applied (for example anything over 90% of duration can be considered a full duration).

Reply:
Makes sense, added a 10% tolerance

Reply:
You're right, missed that it's an integer division, not float. Converting it to float and rounding it off to the nearest integer instead.

Reply:
That could add a dropped packet at 15.1 ms. I am thinking it should add a packet at > 19ms only. So, you would have to include that tolerance in the number of packets calculation too.

Reply:
Hmm okay, let me adjust that